webapp/lib/app/configurator/standard_messages/controller.dart

```diff
@@ -90,76 +95,117 @@ class MessagesConfiguratorController extends ConfiguratorController {
+          for(var group in category.groups.values) {
+            var messageIds = group.messages.keys;
+            if (messageIds.contains(messageData.messageId)) {
+              categoryName = category.categoryName;
+              groupName = group.groupName;
+              break;
+            }
           }
-        });
+        }
+
+        // todo: unwanted map since we use only the category Id, group Id
+        var messageCategoryMap = {
+          standardMessage.categoryId: MessageCategory(standardMessage.categoryId, categoryName)
+            ..groups = {
+              standardMessage.groupId: MessageGroup(messageData.groupId, groupName)
+                ..messages = {
+                  standardMessage.docId: standardMessage
+                }
+            }
+        };
+        _modifyMessagesInView(messageCategoryMap);
         _updateUnsavedIndicators(standardMessagesManager.categories, standardMessagesManager.unsavedMessageIds, standardMessagesManager.unsavedGroupIds, standardMessagesManager.unsavedCategoryIds);
         break;
 
       case MessagesConfigAction.removeStandardMessage:
         StandardMessageData messageData = data;
-        var standardMessage = standardMessagesManager.deleteMessage(messageData.id);
-        _removeMessagesFromView({
-          standardMessage.category: {
-            standardMessage.groupDescription: [standardMessage]
-          }
-        });
+        var standardMessage = standardMessagesManager.deleteMessage(messageData.messageId);
+        var messageCategoryMap = {
+          standardMessage.categoryId: MessageCategory(standardMessage.categoryId, standardMessage.category)
+            ..groups = {
+              standardMessage.groupId: MessageGroup(messageData.groupId, messageData.group)
+                ..messages = {
+                  standardMessage.docId: standardMessage
+                }
+            }
+        };
+        // todo: unwanted map since we use only the category Id, group Id
+        _removeMessagesFromView(messageCategoryMap);
         _updateUnsavedIndicators(standardMessagesManager.categories, standardMessagesManager.unsavedMessageIds, standardMessagesManager.unsavedGroupIds, standardMessagesManager.unsavedCategoryIds);
         break;
 
       case MessagesConfigAction.addStandardMessagesGroup:
         StandardMessagesGroupData groupData = data;
-        var newGroup = standardMessagesManager.createStandardMessagesGroup(groupData.category);
-        _addMessagesToView({groupData.category: {newGroup.groupDescription: []}}, startEditingName: true);
+        var newGroup = standardMessagesManager.createStandardMessagesGroup(groupData.categoryId, groupData.categoryName);
+        var messageCategoryMap = {
+          groupData.categoryId: MessageCategory(groupData.categoryId, groupData.categoryName)
+            ..groups = {
+              newGroup.groupId: MessageGroup(newGroup.groupId ?? model.generateStandardMessageGroupId(), newGroup.groupName)
```

Comment:
Can we have this be:
```suggestion
              newGroup.groupId: MessageGroup(newGroup.groupId, newGroup.groupName)
```
(so relying on the ID generation to be done by `standardMessagesManager.createStandardMessagesGroup`?)